src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsJsonConverter.cs

```
@@ -0,0 +1,142 @@
+        {
+            var problemDetails = new ProblemDetails();
+
+            if (!reader.Read())
+            {
+                throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+            }
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                ReadValue(ref reader, problemDetails, options);
+            }
+
+            return problemDetails;
+        }
+
+        public override void Write(Utf8JsonWriter writer, ProblemDetails value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+            WriteProblemDetails(writer, value, options);
+            writer.WriteEndObject();
+        }
+
+        internal static void ReadValue(ref Utf8JsonReader reader, ProblemDetails value, JsonSerializerOptions options)
+        {
+            if (TryReadStringProperty(ref reader, Type, out var propertyValue))
+            {
+                value.Type = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Title, out propertyValue))
+            {
+                value.Title = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Detail, out propertyValue))
+            {
+                value.Detail = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Instance, out propertyValue))
+            {
+                value.Instance = propertyValue;
+            }
+            else if (reader.ValueTextEquals(Status.EncodedUtf8Bytes))
+            {
+                if (!reader.Read())
+                {
+                    throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+                }
+
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    // Nothing to do here.
+                }
+                else if (reader.TokenType == JsonTokenType.Number)
```

Are we OK with silently ignoring if TokenType is something else or do we need an `else throw`?